packages/gestalt/src/Collage.js

```diff
@@ -0,0 +1,270 @@
+  }
+
+  // If we have a cover image, figure out how big it is, then move all the
+  // existing columns over.
+  if (hasCoverImage) {
+    const coverImageWidth = Math.ceil(numCols / 2) * (width / numCols);
+    items.unshift({ isCover: true });
+    positions = positions.map(position => ({
+      ...position,
+      left: coverImageWidth + position.left,
+    }));
+    positions.unshift({ top: 0, left: 0, width: coverImageWidth, height });
+  }
+
+  // This adds the space between any items that we have. It's nice to do
+  // this as a separate pass after everything else, because the math earlier
+  // becomes easier and it's less brittle to change.
+  if (space) {
+    positions = paddingAll(space, positions);
+  }
+
+  return { items, positions };
+}
+
+type Props = {|
+  height: number,
+  imageAlt?: string,
+  images: Array<ImageType>,
+  imagesAreRounded?: boolean,
+  numCols: number,
+  renderCoverImage?: ({| width: number, height: number |}) => React.Node,
+  space?: CollageSpace,
+  width: number,
+|};
+
+export default function Collage(props: Props) {
+  const {
+    height,
+    imageAlt,
+    images,
+    imagesAreRounded,
+    numCols,
+    renderCoverImage,
+    space,
+    width,
+  } = props;
+  const sizeOffset = space && space.type === 'between' ? space.value * 2 : 0;
+  const { items, positions } = getCollageLayout({
+    width: width + sizeOffset,
+    height: height + sizeOffset,
+    numCols,
+    space,
+    images,
+    hasCoverImage: !!renderCoverImage,
+  });
+  const imageShape = imagesAreRounded ? 'rounded' : 'square';
+  return (
+    <Collection
+      Item={({ idx }) => {
+        const item = items[idx];
+        const position = positions[idx];
+        const { width: itemWidth, height: itemHeight } = position;
+        if (!item) {
+          return (
+            <Box
+              color="lightGray"
+              height={itemHeight}
+              shape={imageShape}
+              width={itemWidth}
+            />
+          );
+        }
+        if (item.isCover && renderCoverImage) {
+          return (
+            <Mask height={itemHeight} shape={imageShape}>
+              {renderCoverImage({ width: itemWidth, height: itemHeight })}
+            </Mask>
+          );
+        }
+        return (
+          <Mask wash height={itemHeight} shape={imageShape}>
+            <Image
+              alt={imageAlt || ''}
+              color={item.color || '#EFEFEF'}
+              fit="cover"
+              naturalHeight={item.naturalHeight || 1}
+              naturalWidth={item.naturalWidth || 1}
+              src={item.src || ''}
+            />
+          </Mask>
+        );
+      }}
+      itemPadding={(space && space.value) || 0}
+      layout={positions}
+    />
+  );
+}
+
+Collage.propTypes = {
+  height: PropTypes.number.isRequired,
+  imageAlt: PropTypes.string,
+  images: PropTypes.arrayOf(ImagePropType),
+  imagesAreRounded: PropTypes.bool,
+  numCols: PropTypes.number.isRequired,
```

Comment:
This might be nicer as just `columns` - `num` is a type hint and is already expressed elsewhere.

Reply:
I know `columns` is a number hinted by flow, but I usually thought of plural as an array of things instead of a number. I know Masonry uses `minCols` so we're already using plural for non-Array, but I feel that's more intuitive because of the adjective `min`. Thoughts?